app/controllers/sessions_controller.rb

```diff
@@ -36,35 +36,30 @@ def auth_hash
   def authorize_access!
     return true unless Octobox.restricted_access_enabled?
 
-    client = Octokit::Client.new(access_token: auth_hash.credentials.token)
-    nickname = auth_hash.info.nickname
-
-    return true if organization_member?(client, nickname) || team_member?(client, nickname)
+    client = Octokit::Client.new(access_token: auth_hash.credentials.token, auto_paginate: true)
+    return true if organization_member?(client) || team_member?(client)
 
     flash[:error] = 'Access denied.'
     redirect_to root_path
   end
 
-  def organization_member?(*args)
-    member?(*args, type: :organization)
-  end
+  def organization_member?(client)
+    org_id = Octobox.config.github_organization_id
+    return false unless org_id
+
+    orgs = client.organizations(headers: { 'Cache-Control' => 'no-cache, no-store' })
+    return false unless orgs
 
-  def team_member?(*args)
-    member?(*args, type: :team)
+    orgs.any? { |org| org['id'].to_i == org_id }
   end
 
-  def member?(client, nickname, type:)
-    id = Octobox.config.public_send("github_#{type}_id")
-    return false unless id.present?
-
-    begin
-      client.public_send("#{type}_member?",
-        id,
-        nickname,
-        headers: { 'Cache-Control' => 'no-cache, no-store' }
-      )
-    rescue Octokit::Error
-      false
-    end
+  def team_member?(client)
+    team_id = Octobox.config.github_team_id
+    return false unless team_id
+
+    teams = client.user_teams(headers: { 'Cache-Control' => 'no-cache, no-store' })
```

Comment:
@jules2689 sorry had a thought after I approved this PR, what do you think about using this API instead?

https://developer.github.com/v3/orgs/teams/#get-team-membership

And then checking to make sure the state is `active`. This would save us from listing through all of the teams that user belongs to.